openmdao/core/driver.py

```diff
@@ -22,7 +22,80 @@
 from openmdao.utils.array_utils import sizes2offsets
 from openmdao.vectors.vector import _full_slice, _flat_full_indexer
 from openmdao.utils.indexer import indexer
-from openmdao.utils.om_warnings import issue_warning, DerivativesWarning, DriverWarning
+from openmdao.utils.om_warnings import issue_warning, DerivativesWarning, \
+    DriverWarning, OMDeprecationWarning
+
+
+class DriverResult():
+    """
+    A container that stores information pertaining to the result of a driver execution.
+
+    Attributes
+    ----------
+    runtime : float
+        The time required to execute the driver, in seconds.
+    iter_count : int
+        The number of iterations used by the optimizer.
+    obj_calls : int
+        The number of times the objective function was evaluated (model solve_nonlinear calls).
+    deriv_calls : int
+        The number of times the total jacobian was computed.
+    exit_status : str
+        A string that may provide more detail about the results of the driver run.
+    success : bool
+        A boolean that dictates whether or not the driver was successful.
+    """
+
+    def __init__(self):
+        """
+        Initialize the DriverResult object.
+        """
+        self.runtime = 0.0
+        self.iter_count = 0
+        self.obj_calls = 0
```

Comment:
What do you think about maybe keeping track of runtime for obj_calls and deriv_calls separately?  Might be nice to know where the time is being spent.